sdk/storage/Azure.Storage.Files.Shares/src/ShareExtensions.cs

```diff
@@ -52,5 +68,85 @@ private static string ToFileDateTimeString(this DateTimeOffset dateTimeOffset)
+            }
+            return result;
+        }
+
+        internal static ShareItem ToShareItem(this ShareItemInternal shareItemInternal)
+        {
+            if (shareItemInternal == null)
+            {
+                return null;
+            }
+
+            return new ShareItem
+            {
+                Name = shareItemInternal.Name,
+                Snapshot = shareItemInternal.Snapshot,
+                IsDeleted = shareItemInternal.IsDeleted,
+                VersionId = shareItemInternal.VersionId,
+                Properties = shareItemInternal.Properties.ToShareProperties()
+            };
+        }
+
+        internal static ShareProperties ToShareProperties(this SharePropertiesInternal sharePropertiesInternal)
+        {
+            if (sharePropertiesInternal == null)
+            {
+                return null;
+            }
+
+            return new ShareProperties
+            {
+                LastModified = sharePropertiesInternal.LastModified,
+                ETag = sharePropertiesInternal.ETag,
+                ProvisionedIops = sharePropertiesInternal.ProvisionedIops,
+                ProvisionedIngressMBps = sharePropertiesInternal.ProvisionedIngressMBps,
+                ProvisionedEgressMBps = sharePropertiesInternal.ProvisionedEgressMBps,
+                NextAllowedQuotaDowngradeTime = sharePropertiesInternal.NextAllowedQuotaDowngradeTime,
+                DeletedOn = sharePropertiesInternal.DeletedOn,
+                RemainingRetentionDays = sharePropertiesInternal.RemainingRetentionDays,
+                AccessTier = sharePropertiesInternal.AccessTier,
+                AccessTierChangeTime = sharePropertiesInternal.AccessTierChangeTime,
+                AccessTierTransitionState = sharePropertiesInternal.AccessTierTransitionState,
+                LeaseStatus = sharePropertiesInternal.LeaseStatus,
+                LeaseState = sharePropertiesInternal.LeaseState,
+                LeaseDuration = sharePropertiesInternal.LeaseDuration,
+                EnabledProtocols = ToShareEnabledProtocols(sharePropertiesInternal.EnabledProtocols),
+                RootSquash = sharePropertiesInternal.RootSquash,
+                QuotaInGB = sharePropertiesInternal.QuotaInGB,
+                Metadata = sharePropertiesInternal.Metadata
+            };
+        }
+
+        internal static ShareEnabledProtocols? ToShareEnabledProtocols(string rawProtocols)
+        {
+            if (rawProtocols == null)
+            {
+                return null;
+            }
+
+            switch (rawProtocols)
+            {
+                case Constants.File.SmbProtocol:
+                    return ShareEnabledProtocols.Smb;
+                case Constants.File.NfsProtocol:
+                    return ShareEnabledProtocols.Nfs;
+                default:
+                    throw new ArgumentException($"Unknown share enabled protocol: {rawProtocols}");
+            }
```

Comment:
should this make an union ?

Reply:
I think it is fine for now, I will update when the service updates.

Reply:
It might be nasty if customer is using this to read and service version moves but they stay on old SDK.
I think SDK should produce union of know protocols and ignore new protocols.

Reply:
ah, they'd lock on service version. nvm.

Reply:
if the customer says on the old SDK, they will stay on the old service version, and won't encounter this issue.